### PR TITLE
Allow clients an opportunity to see what fields are invalid in create…

### PR DESCRIPTION
**Pull request by robstryker**
…Server request

Signed-off-by: Rob Stryker <rob@oxbeef.net>

### CODE REVIEW

**Comment by robstryker** [comment]
testPR

File: client/jars/org.jboss.tools.rsp.client.cli/src/main/java/org/jboss/tools/rsp/client/cli/StandardCommandHandler.java [hunk 1]
```diff
@@ -351,69 +379,87 @@ private void promptForAttributes(Attributes attr, HashMap<String, Object> store,
...
+	private void promptForInvalidAttributes(List<String> invalid, Attributes attr, HashMap<String, Object> store, boolean required2) {
+		if (attr == null)
+			return;
+		
+		CreateServerAttributesUtility attrsUtil = new CreateServerAttributesUtility(attr);
+		HashMap<String, Object> toSend = store;
+		if (attrsUtil != null) {
+			Set<String> keys = attrsUtil.listAttributes();
+			for (String k : keys) {
+				promptForAttributeSingleKey(attrsUtil, k, required2, toSend);
```
[the remark this answers is not in the document]
Comment (robstryker) [reply client/jars/org.jboss.tools.rsp.client.cli/src/main/java/org/jboss/tools/rsp/client/cli/StandardCommandHandler.java]: I hate one-liners, especially if they're hiding larger logic like a prompt to the user.

File: client/jars/org.jboss.tools.rsp.client.cli/src/main/java/org/jboss/tools/rsp/client/cli/StandardCommandHandler.java [hunk 2]
```diff
@@ -330,18 +331,45 @@ private void runAddServer() {
...
+		System.out.println("Error adding server: " + result.getStatus().getMessage());
+		List<String> list = result.getInvalidKeys();
+		System.out.println("Invalid attributes: ");
+		for( int i = 0; i < list.size(); i++ ) {
+			System.out.println("   " + list.get(i));
+		}
+		
+		System.out.println("Would you like to correct the invalid fields and try again? y/n");
+		String tryAgain = nextLine();
+		if( tryAgain.equalsIgnoreCase("y") || tryAgain.equalsIgnoreCase("yes")) {
```
[the remark this answers is not in the document]
Comment (robstryker) [reply client/jars/org.jboss.tools.rsp.client.cli/src/main/java/org/jboss/tools/rsp/client/cli/StandardCommandHandler.java]: Fixing it

**Comment by adietish** [comment]
One more cleanup here: https://github.com/robstryker/rsp-server/pull/6

**Pull request merged**